### PR TITLE
Previous server connection history

### PR DESCRIPTION
**Pull request by poixen**
newlogin

+ able to select previous servers from a list to connect to.
+ remove items from the drop down by selecting them and pressing delete.
+ new hosts will be added to the previous hosts.
+ remembers the last host connected to in the dropdown.
+ default hosts are "cockatrice.woogerworks.com", "vps.poixen.com" and "chickatrice.net"

### CODE REVIEW

**Comment by Daenyth** [comment]
Instead of previous host/ new host radio buttons, could we do a dropdown with an editable field for adding new items? Does Qt have such a widget?

Do we want to include chickatrice?

**Comment by Daenyth** [comment]
How does password saving work with the host dropdown btw?

**Comment by ZeldaZach** [comment]
I have a feeling that passwords are kept in plaintext if they're saved with the dropdowns.

**Comment by Daenyth** [comment]
They've always been kept in plaintext, see #344

**Comment by ZeldaZach** [comment]
Is that a bad thing?

**Comment by Daenyth** [comment]
Arguably yes or no. Details are in that ticket.

**Comment by ctrlaltca** [comment]
Do we want per-server user profile or just multiple server entries?

**Comment by ZeldaZach** [comment]
Do you mean like if I have two accounts I'd have two options in the drop down menu?

**Comment by ctrlaltca** [comment]
I meant: do we just want to save a list of recent servers, or do we want to have a list of profiles, where each profile has a different server name / user name / password saved?

**Comment by ZeldaZach** [comment]
I like the idea of having profiles

**Comment by poixen** [comment]
@Daenyth that was the first thing I thought of too. Whilst you can have an editable dropdown, it makes all of the items editable - its not good. I couldnt find anything else, so this was the next best thing.

Profiles would be great! If someone wants to expand on this they can. I wont be adding more to this PR  :/

**Comment by poixen** [comment]
Oh and I though about adding chickatrice, but I couldnt connect. Are they reg only?

**Comment by MassCraziness** [comment]
Yes, they are reg-only @poixen

**Comment by poixen** [comment]
Added chickatrice. ready to be tested.

**Comment by Daenyth** [comment]
Profiles were what I had in mind. Users should use different passwords on
different servers, and this encourages you to send your password for server
A to server B.

**Comment by Daenyth** [comment]
poixen if you don't want to write that I'll try to find some time to work
on it myself, maybe this weekend or something

**Comment by poixen** [comment]
@Daenyth go for it. You can use this as a base if you want.

**Comment by poixen** [comment]
@Daenyth any updates on this?

**Comment by Daenyth** [comment]
Haven't had time to work on it.

File: cockatrice/src/dlg_connect.cpp [hunk 1]
```diff
@@ -109,3 +173,17 @@ void DlgConnect::actCancel()
     
     reject();
 }
+
+
+bool DeleteHighlightedItemWhenShiftDelPressedEventFilter::eventFilter(QObject *obj, QEvent *event)
```
Comment (Daenyth) [inline cockatrice/src/dlg_connect.cpp]: That is quite the class name!

Comment (poixen) [inline cockatrice/src/dlg_connect.cpp]: Ha! Forgot about that. It's some net code I found.

**Comment by tooomm** [comment]
@Lachee any knews about this? Just saw it on your chickatrice.net site:
chickatrice net adress

Makes no sense to implement a dead address. ;)

**Comment by Lachee** [comment]
I forgot I placed that. My domain seller.triednto scare me by saying the chickatrice.net was about to expire on the 3rd of August, but obviously it didn't. As I am a tad broke at the moment, I wasn't sure if I should continue the domain or buy voidpixel.com.an. 

I since learnt that the domain doesn't expire until the 13 of October or something like that, so I decided to put it of for a while and forgot to take down the messages.

I shall do that today, thank you for reminding me ;)

Cheers, Lachee

Date: Wed, 5 Aug 2015 07:39:54 -0700

**Comment by ZeldaZach** [comment]
Is anything being done with this currently?

**Comment by poixen** [comment]
This PR is good to go. Not sure if @Daenyth is working on another PR based on it.

**Comment by ZeldaZach** [comment]
I'll let @Daenyth merge it if he wants then

**Comment by Daenyth** [comment]
:shipit:, I'm not going to have time for a little bit and I don't want to hold this up. I'll get my enhancements in later.

**Pull request merged**